vm/src/core/mod.rs

```diff
@@ -18,7 +18,7 @@ macro_rules! assert_deq {
         match (&($left), &($right)) {
             (left_val, right_val) => {
                 if !(*left_val == *right_val) {
-                    panic!("assertion failed: `(left == right)` \
+                    ice!("assertion failed: `(left == right)` \
```

Comment:
This assert is only used in tests so they aren't really an ice and could remain as panic.